### ISSUE TITLE
NameError: global name 'time' is not defined

### ISSUE THREAD

**Issue by JohnLahr** [Tautulli/Tautulli#597]
I noticed the "History" tab never showed info and instead just continued to display "fetching rows", so I thought I'd restart PlexPy in an attempt to resolve that.  However, upon restarting, PlexPy just crashes with this in the log:

```
06-Mar-2016 17:48:31 - INFO    :: MainThread : Checking to see if the database has all tables....
06-Mar-2016 17:48:31 - INFO    :: MainThread : Retrieving latest version information from GitHub
06-Mar-2016 17:48:31 - DEBUG   :: MainThread : Requesting URL via GET method: https://api.github.com/repos/drzoidberg33/plexpy/commits/master
06-Mar-2016 17:48:32 - DEBUG   :: MainThread : Latest version is 65429975203f64a3b524a6e33b8584d4c0e20e9e
06-Mar-2016 17:48:32 - INFO    :: MainThread : PlexPy is up to date
06-Mar-2016 17:48:32 - INFO    :: MainThread : PlexPy PlexTV :: Requesting URLs for server...
06-Mar-2016 17:48:32 - INFO    :: MainThread : Writing configuration to file
06-Mar-2016 17:48:33 - INFO    :: MainThread : Writing configuration to file
06-Mar-2016 17:48:33 - INFO    :: MainThread : PlexPy Pmsconnect :: Requesting name from server...
06-Mar-2016 17:48:33 - INFO    :: MainThread : PlexPy PlexTV :: Requesting users list refresh...
06-Mar-2016 17:48:35 - WARNING :: MainThread : PlexPy Database :: Database Error: unable to open database file
06-Mar-2016 17:48:35 - ERROR   :: MainThread : Uncaught exception: Traceback (most recent call last):
  File "C:\Websites\PlexPy\PlexPy.py", line 232, in <module>
    main()
  File "C:\Websites\PlexPy\PlexPy.py", line 160, in main
    plexpy.initialize(config_file)
  File "C:\Websites\PlexPy\plexpy\__init__.py", line 188, in initialize
    plextv.refresh_users()
  File "C:\Websites\PlexPy\plexpy\plextv.py", line 56, in refresh_users
    monitor_db.upsert('users', new_value_dict, control_value_dict)
  File "C:\Websites\PlexPy\plexpy\database.py", line 178, in upsert
    self.action(update_query, value_dict.values() + key_dict.values())
  File "C:\Websites\PlexPy\plexpy\database.py", line 139, in action
    time.sleep(1)
NameError: global name 'time' is not defined
```

I tried downloading a new copy of PlexPy, but with a fresh config and everything, it can no longer communicate with my Plex Server in the same subnet. If I manually change the config.ini settings to match my old one, I get the same error above.

**Comment by JohnLahr** [comment]
I'm now noticing that even fresh copies of PlexPy will crash after the first restart, regardless of whether any settings are changed.

**Comment by JonnyWong16** [comment]
Are you trying to use your old `plexpy.db` file? It looks like there is something wrong with it.

**Comment by JohnLahr** [comment]
I was. However, a completely fresh copy of PlexPy is still breaking after a restart. I'm not using any old files, changing any settings, etc. Just installing, running the first-time setup, then restarting it, and it throws me this error again.

**Comment by JonnyWong16** [comment]
Can you provide all this info? https://github.com/drzoidberg33/plexpy#if-nothing-has-worked

Maybe permission issues preventing PlexPy from opening the database?

**Comment by JohnLahr** [comment]
Version: Latest
Branch: Master
Commit: 65429975203f64a3b524a6e33b8584d4c0e20e9e
OS: Windows Server 2012 R2
Python: 2.7.10 (also tried 2.7.11 with no luck)

I'm not using my old database file or config. I completely started fresh. The only thing I've entered is my Plex user and server info at the initial setup. Without changing any other settings, if I restart the PlexPy app, it crashes with that error upon launch.

**Comment by JonnyWong16** [comment]
Ok, I found the 'time' bug, but your underlying issue is still this:

```
PlexPy Database :: Database Error: unable to open database file
```

**Comment by JohnLahr** [comment]
It's trying to open the newly created empty database from the initial setup. I've even granted "Everyone" full access to the file. I'm not sure why it isn't able to open it, especially since it's a fresh file with full access permissions.

**Comment by JonnyWong16** [comment]
How are you restarting PlexPy?

**Comment by JohnLahr** [comment]
Through the web interface, under "Settings".

**Comment by JonnyWong16** [comment]
Can you PM me your database file on the [Plex forums](http://forums.plex.tv/profile/JonnyWong16)? Both your old db file and the new clean one.

**Comment by JohnLahr** [comment]
Actually, I uninstalled Python and installed the 64-bit version, and for whatever reason, it can read the database now. I'm not sure why that mattered, but it works now. Thanks for taking the time to help even though it looks like the fault was on my end.

**Comment by JonnyWong16** [comment]
Ok, glad you figured it out. 

But thanks for noticing the 'time' thing. It's a bug that needs to be fixed.

**Comment by JonnyWong16** [comment]
Fixed in v1.3.10.
